packages/aws-rfdk/lib/deadline/lib/worker-configuration.ts

```diff
@@ -29,6 +29,26 @@ import {
   Version,
 } from './version';
 
+/**
+ * Provider for adding user data scripts
+ */
+export interface IInstanceUserDataProvider {
+  // Method that is invoked before configuration Cloud Watch Agent.
+  preCloudWatchAgent(host: IHost): void;
+  // Method that is invoked before Deadline worker and renderqueue configuration .
+  preDeadlineConfiguration(host: IHost): void;
+  // Method that is invoked after all configuration is done and worker started.
+  postWorkerLaunch(host: IHost): void;
+}
+
+export class InstanceUserDataProvider implements IInstanceUserDataProvider{
+  preCloudWatchAgent(_host: IHost): void {
+  }
+  preDeadlineConfiguration(_host: IHost): void {
+  }
+  postWorkerLaunch(_host: IHost): void {
+  }
```

Comment:
nit: we should have a blank line between these functions to be consistent with the rest of the codebase. Same applies elsewhere